edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/AuthAgentProtocolHead.cs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    using Microsoft.AspNetCore.Hosting;
+
+    using Microsoft.Azure.Devices.Edge.Hub.Core;
+    using Microsoft.Azure.Devices.Edge.Hub.Core.Identity;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    using Microsoft.Extensions.Logging;
+
+    public class AuthAgentProtocolHead : IProtocolHead
+    {
+        readonly IAuthenticator authenticator;
+        readonly IUsernameParser usernameParser;
+        readonly IClientCredentialsFactory clientCredentialsFactory;
+        readonly AuthAgentProtocolHeadConfig config;
+
+        private IWebHost host;
```

Comment:
Nit: remove private.

Reply:
Also, to avoid nulls, can we use Option and locking instead?

Reply:
changed to Option - however using null/exchange was cleaner I think.